src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/InheritDocTaglet.java

```diff
@@ -70,25 +73,72 @@ public InheritDocTaglet() {
      */
     private Content retrieveInheritedDocumentation(TagletWriter writer,
                                                    ExecutableElement method,
-                                                   DocTree inheritDoc,
+                                                   InheritDocTree inheritDoc,
                                                    boolean isFirstSentence) {
         Content replacement = writer.getOutputInstance();
         BaseConfiguration configuration = writer.configuration();
         Messages messages = configuration.getMessages();
         Utils utils = configuration.utils;
         CommentHelper ch = utils.getCommentHelper(method);
-        var path = ch.getDocTreePath(inheritDoc).getParentPath();
+        DocTreePath inheritDocPath = ch.getDocTreePath(inheritDoc);
+        var path = inheritDocPath.getParentPath();
         DocTree holderTag = path.getLeaf();
+
+        ExecutableElement src = null;
+        // 1. Does @inheritDoc specify a type?
+        if (inheritDoc.getSupertype() != null) {
+            // 2. Can we find that type?
+            var supertype = (TypeElement) ch.getReferencedElement(inheritDoc.getSupertype());
+            if (supertype == null) {
+                messages.error(inheritDocPath, "doclet.inheritDocBadSupertype");
+                return replacement;
```

Comment:
`replacement` may be empty.
Consider using `HtmlDocletWriter.invalidTagOutput`.

(Later, we might want to provide a utility message to wrap `messages.error` and `invalidTagOutput`